packages/ui/src/lib/table/Table.svelte

```
@@ -249,73 +246,48 @@ function sort(column: Column<T>): void {
+});
 
-function objectChecked(object: T): void {
-  // check for children and set them to the same state
+function objectChecked(object: T, checked: boolean): void {
+  object.selected = checked;
   if (row.info.children) {
     const children = row.info.children(object);
     if (children) {
-      // event fires before parent changes so use '!'
-      children.forEach(child => (child.selected = !object.selected));
+      children.forEach(child => (child.selected = checked));
```

Biome (2.4.16)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)